source/config.hpp

```diff
@@ -50,6 +50,7 @@ class Config
 	string default_member_lvalue_reference_return_value_policy_ = "pybind11::return_value_policy::automatic";
 	string default_member_rvalue_reference_return_value_policy_ = "pybind11::return_value_policy::automatic";
 	string default_call_guard_ = "";
+	string custom_shared_ = "";
```

Comment:
Optional: It feel like this should have default `std::shared_ptr`, but then whole name of this option became a bit wrong, perhaps something like `holder_type` would be a bit more accurate, thoughts?

Reply:
Yes and that would simplify this:
```
std::string shared_ptr_class_name() {
  return shared_ptr = Config::get().use_custom_shared() ? Config::get().custom_shared() : "std::shared_ptr"; 
}
```
Instead, we could just use:
`Config::get().holder_type()`
with 
`string const &holder_type() const { return holder_type_; }`

Actually, with all of the above comments, we can even get rid of the function `use_custom_shared()`.